drivers/javascript/errors.coffee

```diff
@@ -24,9 +24,19 @@ class RqlServerError extends Error
 class RqlRuntimeError extends RqlServerError
 
 class RqlCompileError extends RqlServerError
-
 class RqlClientError extends RqlServerError
 
+class RqlInternalError extends RqlRuntimeError
+class RqlResourceError extends RqlRuntimeError
+class RqlLogicError extends RqlRuntimeError
+class RqlNonExistenceError extends RqlRuntimeError
```

Comment:
A `NonExistenceError` is a type of `LogicError` (currently it's the only type we report on its own), so it should probably extend `RqlLogicError` instead.